beacon-chain/state/types_test.go

```diff
@@ -0,0 +1,144 @@
+package state
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+
+	"github.com/prysmaticlabs/prysm/shared/interop"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/stateutil"
+)
+
+func TestBeaconState_ProtoBeaconStateCompatibility(t *testing.T) {
```

Comment:
Should you maybe add a test to make sure it's equal to the proto.Clone of beacon state?